src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -11,135 +11,141 @@ namespace System.Text.RegularExpressions
-        public System.Text.RegularExpressions.Match NextMatch() { return default(System.Text.RegularExpressions.Match); }
-        public virtual string Result(string replacement) { return default(string); }
+        public static System.Text.RegularExpressions.Match Empty { get { throw null; } }
+        public virtual System.Text.RegularExpressions.GroupCollection Groups { get { throw null; } }
+        public System.Text.RegularExpressions.Match NextMatch() { throw null; }
+        public virtual string Result(string replacement) { throw null; }
+        public static System.Text.RegularExpressions.Match Synchronized(System.Text.RegularExpressions.Match inner) { throw null; }
     }
     public partial class MatchCollection : System.Collections.ICollection, System.Collections.IEnumerable
     {
         internal MatchCollection() { }
-        public int Count { get { return default(int); } }
-        public virtual System.Text.RegularExpressions.Match this[int i] { get { return default(System.Text.RegularExpressions.Match); } }
-        bool System.Collections.ICollection.IsSynchronized { get { return default(bool); } }
-        object System.Collections.ICollection.SyncRoot { get { return default(object); } }
-        public System.Collections.IEnumerator GetEnumerator() { return default(System.Collections.IEnumerator); }
-        void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
+        public int Count { get { throw null; } }
+        public bool IsReadOnly { get { throw null; } }
+        public bool IsSynchronized { get { throw null; } }
+        public virtual System.Text.RegularExpressions.Match this[int i] { get { throw null; } }
+        public object SyncRoot { get { throw null; } }
+        public void CopyTo(System.Array array, int arrayIndex) { }
+        public System.Collections.IEnumerator GetEnumerator() { throw null; }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
-    public abstract partial class RegexRunnerFactory
-    {
-        protected RegexRunnerFactory() { }
-        protected internal abstract RegexRunner CreateInstance();
-    }
-    public partial class Regex
+    public partial class Regex : System.Runtime.Serialization.ISerializable
     {
-        public static readonly System.TimeSpan InfiniteMatchTimeout;
-        protected internal string pattern;
-        protected internal RegexRunnerFactory factory;
-        protected internal RegexOptions roptions;
-        protected internal TimeSpan internalMatchTimeout;
+        protected internal System.Collections.Hashtable caps;
+        protected internal System.Collections.Hashtable capnames;
         protected internal int capsize;
         protected internal string[] capslist;
-        protected System.Collections.IDictionary Caps { get; set; }
```

Comment:
Caps and CapNames should remain in the ref for now pending the outcome of issue I asked you to file about these.